Dockerfile

```diff
@@ -456,9 +457,10 @@ COPY --from=unit-tests-runner /src/coverage.txt /coverage.txt
 FROM base AS unit-tests-race
 RUN unlink /etc/ssl
 COPY --from=rootfs / /
-COPY hack/golang/test.sh /bin
 ARG TESTPKGS
-RUN --security=insecure --mount=type=cache,id=testspace,target=/tmp --mount=type=cache,target=/.cache/go-build /bin/test.sh --race ${TESTPKGS}
+ENV PLATFORM container
+ENV CGO_ENABLED 1
+RUN --security=insecure --mount=type=cache,id=testspace,target=/tmp --mount=type=cache,target=/.cache/go-build --mount=type=cache,target=/go/pkg go test -v -race -count 1 -p 4 ${TESTPKGS}
```

Comment:
Why we pass `-p 4` explicitly?

Reply:
because otherwise it eats all the CPUs available on the node which kills the CI node or your computer; our unit-tests are not truly unit, some launch containerd, etc., which introduces more load than a node can handle. `-p 4` was a compromise